packages/cubejs-schema-compiler/adapter/BaseQuery.js

```diff
@@ -952,17 +960,27 @@ class BaseQuery {
           this.safeEvaluateSymbolContext().leafMeasures[this.safeEvaluateSymbolContext().currentMeasure] = true;
         }
       }
+      let evaluatedSql;
+      if (this.safeEvaluateSymbolContext().rollupQuery && symbol.type === 'sum' && symbol.filters) {
```

Comment:
@ferrants Great! I think we're almost there! I believe we don't need these changes if we add timeDimensions to `renderedReference` as well. We should also get rid of these lines then:
https://github.com/cube-js/cube.js/blob/4705af080cd9549ff735f3ab6ea99dfd8de8b4ac/packages/cubejs-schema-compiler/adapter/BaseQuery.js#L894-L897

Reply:
This didn't work. I did add the `timeDimensions`, but the dimension, `source` I'm using for this measure filter in the tests isn't a timeDimension. Also, there is an issue with nested aggregate functions if I remove these lines. I end up with a:
```
sum(case when "visitors__source" = 'google' then sum("visitors__count") end)
```